### PR TITLE
Created database list (task #2417)

### PR DESCRIPTION
**Pull request by flangofas**
### Database list feature:

Let's say from the UI, we have created **contact-types** list and its list items.

Then, in the module's config file we use it as follows:

~~~
FIELD NAME,FIELD TYPE,REQUIRED,NOT SEARCHABLE,UNIQUE
id,uuid
type,dblist(contact-types),
~~~

### What it means:

From the example above, type field is select input field and its options are fetched from the list contact-types. This list must be created by the end-user on the UI first.

### CODE REVIEW

File: config/Migrations/20160830084634_CreateDbLists.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+     * @return void
+     */
+    public function change()
+    {
+        $table = $this->table('db_lists', ['id' => false, 'primary_key' => ['id']]);
+        $table->addColumn('id', 'uuid', [
+            'default' => null,
+            'null' => false,
+        ]);
+        $table->addColumn('name', 'string', [
```
Comment (mamchenkov) [inline config/Migrations/20160830084634_CreateDbLists.php]: @flangofas list name should probably also be unique

File: config/Migrations/20160830084645_CreateDbListItems.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            'default' => null,
+            'null' => false,
+            'limit' => 255,
+        ]);
+        $table->addColumn('value', 'string', [
+            'default' => null,
+            'null' => false,
+            'limit' => 255,
+        ]);
+        $table->addColumn('created', 'datetime', [
```
Comment (mamchenkov) [inline config/Migrations/20160830084645_CreateDbListItems.php]: @flangofas list items should also probably have the order column, to show them in the correct order (sorted by).

File: src/FieldHandlers/DbListFieldHandler.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+{
+    /**
+     * Field type
+     */
+    const FIELD_TYPE = 'string';
+
+    /**
+     * Field type match pattern
+     */
+    const FIELD_TYPE_PATTERN = '/list\((.*?)\)/';
```
Comment (mamchenkov) [inline src/FieldHandlers/DbListFieldHandler.php]: @flangofas this field type pattern is already used for the CSV lists.  You need 'dblist' instead.

File: config/Migrations/20160830084645_CreateDbListItems.php (hunk 2 above)
Comment (flangofas) [inline config/Migrations/20160830084645_CreateDbListItems.php]: @mamchenkov at the moment, they are sorted by name in ascending order.

File: config/Migrations/20160830084634_CreateDbLists.php (hunk 1 above)
Comment (flangofas) [inline config/Migrations/20160830084634_CreateDbLists.php]: @mamchenkov done.

**Comment by mamchenkov** [comment]
@flangofas I think we should create the database lists automatically, if they don't already exist.  The admin who will manage the lists does not necessarily know which lists should exist.  So if create them, then it's much easier for him to handle the values.

**Comment by mamchenkov** [comment]
@flangofas I've tested it a bit on my local and it seems to work.  But it'll need more effort.  We should probably merge this PR and do the other work in the additional ones.  Some of the things I've noticed were:

* List value has to be unique.  Otherwise I can add multiple items with the same value and same/different name.
* In the flat list, name has to be unique as well.
* Nested lists are not supported (we do support them in the CSV lists though)
* Active/inactive flag is needed for list items, so that they could be disabled without removing
* The UI needs work.

Waiting for the unit tests from you.

**Pull request merged**